drivers/i3c/i3c_npcx.c

```diff
@@ -2144,41 +2893,53 @@ static int npcx_i3c_init(const struct device *dev)
 		return ret;
 	}
 
+	/* Lock initial */
 	k_mutex_init(&data->lock_mutex);
 	k_sem_init(&data->sync_sem, 0, 1);
 	k_sem_init(&data->ibi_lock_sem, 1, 1);
+	k_sem_init(&data->target_lock_sem, 1, 1);
+	k_sem_init(&data->target_event_lock_sem, 1, 1);
 
 	ret = i3c_addr_slots_init(dev);
 	if (ret != 0) {
 		LOG_ERR("Addr slots init fail %d", ret);
 		return ret;
 	}
 
-	ctrl_config->is_secondary = false; /* Currently can only act as primary controller. */
-	ctrl_config->supported_hdr = I3C_MSG_HDR_DDR; /* HDR-DDR mode is supported. */
-	ctrl_config->scl.i3c = config->clocks.i3c_pp_scl_hz; /* Set I3C frequency */
+	/* Set controller default configuration */
+	config_cntlr->is_secondary = config->secondary_mode;
+	config_cntlr->supported_hdr = I3C_MSG_HDR_DDR;        /* HDR-DDR mode is supported. */
+	config_cntlr->scl.i3c = config->clocks.i3c_pp_scl_hz; /* Set I3C frequency */
 
-	ret = npcx_i3c_configure(dev, I3C_CONFIG_CONTROLLER, ctrl_config);
+	/* Initial the device as controller or target */
+	config_type = (config_target->enable | config_cntlr->is_secondary) ? I3C_CONFIG_TARGET
+									   : I3C_CONFIG_CONTROLLER;
+	config_ptr = (config_type == I3C_CONFIG_CONTROLLER) ? (void *)config_cntlr
+							    : (void *)config_target;
```

Comment:
This still does not make sense to me. If someone has `is_secondary` set to true and `config_target->enable` set to false. Then this would not be a valid startup condition for I3C according to the spec because as a secondary controller you must startup as a target.
I'd suggest remove this `target_mode_en` prop and only use `is_secondary`. If a device is to be seen as a "Target-Only" device, then you are to set the proper values in the BCR so another controller won't try to do a handoff to the device.